main.py

```diff
@@ -373,11 +376,26 @@ def handle_interface_link_up(self, event):
         """
         interface = event.content['interface']
         link = self._get_link_from_interface(interface)
-        if link and not link.is_active():
+        if not link:
+            return
+        if link.endpoint_a == interface:
+            other_interface = link.endpoint_b
+        else:
+            other_interface = link.endpoint_a
+        interface.activate()
+        if other_interface.is_active() is False:
+            return
+        if link.is_active() is False:
+            link.update_metadata('last_status_change', time.time())
             link.activate()
-            self.notify_topology_update()
-            self.update_instance_metadata(interface.link)
-            self.notify_link_status_change(link)
+            time.sleep(self.link_up_timer)
```

Comment:
How can we make sure the NApp thread won't be locked for the time of this sleep?